src/Discord.Net.Core/Entities/Interactions/Modals/ModalBuilder.cs

```diff
@@ -108,21 +110,100 @@ public ModalBuilder AddComponents(List<IMessageComponent> components, int row)
             return this;
         }
 
+        /// <summary>
+        ///     Gets a <see cref="TMessageComponent"/> by the specified <paramref name="customId"/>.
+        /// </summary>
+        /// <typeparam name="TMessageComponent">The type of the component to get.</typeparam>
+        /// <param name="customId">The <see cref="IMessageComponent.CustomId"/> of the component to get.</param>
+        /// <returns>The component of type <typeparamref name="TMessageComponent"/> that was found.</returns>
+        /// <exception cref="ArgumentException">
+        ///     Thrown when the <typeparamref name="TMessageComponent"/> to be updated was not found.
+        /// </exception>
+        public TMessageComponent GetComponent<TMessageComponent>(string customId) 
+            where TMessageComponent : class, IMessageComponent 
+            => Components.ActionRows?.SelectMany(r => r.Components.OfType<TMessageComponent>()).FirstOrDefault(c => c?.CustomId == customId)
+                ?? throw new ArgumentException($"There is no component of type {typeof(TMessageComponent).Name} with the specified custom ID in this modal builder.", nameof(customId));
```

Comment:
@quinchs should this just return a `null` value or throw an exception?

Reply:
also, the wording should be adjusted
> to be updated
this method can be used for different stuff other than updating a component

Reply:
this will return null now (https://github.com/discord-net/Discord.Net/pull/2773/commits/5e2da095bce00ea1409f8661ea8b2866785b53b7)